lib/core/init.d.ts

```diff
@@ -133,4 +132,4 @@ export interface InitOptions extends ServerUrlOptions, ServerInitOptions {
  * @param options of configuration, often these are hardcoded values of the mobile client app.
  * @return promise resolving to Information object as soon as the device is ready.
  */
-export declare function init(options?: InitOptions): Q.Promise<device.Information>;
+export declare function init(options?: InitOptions): Q.Promise<{}>;
```

Comment:
You didn't want to change the return type of init(), did you? See, the compiler dropped the device.Information type here. The result is that users will lose auto-completion of the then function argument when chaining the init promise result like so:

```
Relution.init({}).then((info) => {
   // here compiler no longer knows info is of type device.Information!
});
```

The reason why this is so is because in the init.ts file the return type of init() is not declared explicitly but infered by flow analysis. The flow includes your new call to Q.reject(...) which does not specify a <type> argument, so that the type the promise returned by Q.reject resolves to is unknown. Put differently, a call to Q.reject(...) lacks always lacks important type information!

The compiler needs your help here, please change your code as follows so that the return type of init() can be infered properly:
```
    return Q.reject<device.Information>(new Error...);
```

Giving it the type in the Q.reject() call will make it return a Q.Promise<device.Information> again.